src/vehicle.cpp

```diff
@@ -1344,111 +1345,23 @@ bool vehicle::can_mount( const point &dp, const vpart_id &id ) const
         return false;
     }
 
-    // Alternators must be installed on a gas engine
-    if( part.has_flag( VPFLAG_ALTERNATOR ) ) {
-        bool anchor_found = false;
-        for( const auto &elem : parts_in_square ) {
-            if( part_info( elem ).has_flag( "E_ALTERNATOR" ) ) {
-                anchor_found = true;
-            }
-        }
-        if( !anchor_found ) {
-            return false;
-        }
-    }
-
-    //Seatbelts must be installed on a seat
-    if( part.has_flag( "SEATBELT" ) ) {
-        bool anchor_found = false;
-        for( const auto &elem : parts_in_square ) {
-            if( part_info( elem ).has_flag( "BELTABLE" ) ) {
-                anchor_found = true;
-            }
-        }
-        if( !anchor_found ) {
-            return false;
-        }
-    }
-
-    //Internal must be installed into a cargo area.
-    if( part.has_flag( "INTERNAL" ) ) {
-        bool anchor_found = false;
-        for( const auto &elem : parts_in_square ) {
-            if( part_info( elem ).has_flag( "CARGO" ) ) {
-                anchor_found = true;
-            }
-        }
-        if( !anchor_found ) {
-            return false;
-        }
-    }
-
-    // curtains must be installed on (reinforced)windshields
-    // TODO: do this automatically using "location":"on_mountpoint"
-    if( part.has_flag( "WINDOW_CURTAIN" ) ) {
-        bool anchor_found = false;
-        for( const auto &elem : parts_in_square ) {
-            if( part_info( elem ).has_flag( "WINDOW" ) ) {
-                anchor_found = true;
-            }
-        }
-        if( !anchor_found ) {
-            return false;
-        }
-    }
-
-    // Security system must be installed on controls
-    if( part.has_flag( "ON_CONTROLS" ) ) {
-        bool anchor_found = false;
-        for( auto it : parts_in_square ) {
-            if( part_info( it ).has_flag( "CONTROLS" ) ) {
-                anchor_found = true;
+    // Check all the flags of the part to see if they require other flags
+    // If other flags are required check if those flags are present
+    for( const std::string flag : part.get_flags() ) {
+        if( json_flag::get( flag ).requires_flag() != "" ) {
```

Comment:
```suggestion
        if( !json_flag::get( flag ).requires_flag().empty() ) {
```
But better would be to change `requires_flag_` to a `cata::optional<std::string>`.